Index: ScriptingModules/CEGUILua/tolua++bin/Makefile.am
===================================================================
--- ScriptingModules/CEGUILua/tolua++bin/Makefile.am	(revision 1465)
+++ ScriptingModules/CEGUILua/tolua++bin/Makefile.am	(working copy)
@@ -30,4 +30,4 @@
              lua/verbatim.lua
 
 tolua__cegui_LDFLAGS = -L$(top_builddir)/ScriptingModules/CEGUILua/tolua++/.libs
-tolua__cegui_LDADD = -lCEGUItoluapp
+tolua__cegui_LDADD = -lCEGUItoluapp -lCEGUIBase
Index: ScriptingModules/CEGUILua/tolua++/Makefile.am
===================================================================
--- ScriptingModules/CEGUILua/tolua++/Makefile.am	(revision 1465)
+++ ScriptingModules/CEGUILua/tolua++/Makefile.am	(working copy)
@@ -9,7 +9,7 @@
                              tolua_push.c \
                              tolua_to.c
 
-libCEGUItoluapp_la_LIBADD = $(Lua_LIBS)
+libCEGUItoluapp_la_LIBADD = $(Lua_LIBS) -lCEGUIBase
 
 noinst_HEADERS = tolua_event.h
 pkginclude_HEADERS = tolua++.h
Index: ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.am
===================================================================
--- ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.am	(revision 1465)
+++ ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.am	(working copy)
@@ -4,6 +4,7 @@
 lib_LTLIBRARIES = libCEGUILuaScriptModule.la
 
 libCEGUILuaScriptModule_la_LDFLAGS = -version-info @CEGUI_VERSION_INFO@
+libCEGUILuaScriptModule_la_LIBADD = -lCEGUIBase
 
 libCEGUILuaScriptModule_la_SOURCES = lua_CEGUI.cpp \
                                      CEGUILua.cpp \
Index: WindowRendererSets/Falagard/src/Makefile.am
===================================================================
--- WindowRendererSets/Falagard/src/Makefile.am	(revision 1465)
+++ WindowRendererSets/Falagard/src/Makefile.am	(working copy)
@@ -4,6 +4,7 @@
 lib_LTLIBRARIES=libCEGUIFalagardWRBase.la
 
 libCEGUIFalagardWRBase_la_LDFLAGS = -version-info @CEGUI_VERSION_INFO@
+libCEGUIFalagardWRBase_la_LIBADD = -lCEGUIBase
 
 libCEGUIFalagardWRBase_la_SOURCES = FalButton.cpp \
                                   FalToggleButton.cpp \
Index: RendererModules/OpenGLGUIRenderer/Makefile.am
===================================================================
--- RendererModules/OpenGLGUIRenderer/Makefile.am	(revision 1465)
+++ RendererModules/OpenGLGUIRenderer/Makefile.am	(working copy)
@@ -4,6 +4,6 @@
 lib_LTLIBRARIES=libCEGUIOpenGLRenderer.la
 libCEGUIOpenGLRenderer_la_LDFLAGS = -version-info @CEGUI_OPENGL_VERSION_INFO@
 libCEGUIOpenGLRenderer_la_SOURCES = openglrenderer.cpp opengltexture.cpp
+libCEGUIOpenGLRenderer_la_LIBADD = -lCEGUIBase
 
 
-
Index: RendererModules/IrrlichtRenderer/Makefile.am
===================================================================
--- RendererModules/IrrlichtRenderer/Makefile.am	(revision 1465)
+++ RendererModules/IrrlichtRenderer/Makefile.am	(working copy)
@@ -19,4 +19,4 @@
                                       IrrlichtMemoryFile.cpp
 
 
-libCEGUIIrrlichtRenderer_la_LIBADD = $(Irrlicht_LIBS)
+libCEGUIIrrlichtRenderer_la_LIBADD = $(Irrlicht_LIBS) -lCEGUIBase
Index: Samples/common/src/Makefile.am
===================================================================
--- Samples/common/src/Makefile.am	(revision 1465)
+++ Samples/common/src/Makefile.am	(working copy)
@@ -8,7 +8,7 @@
                                   CEGuiSample.cpp \
                                   CLICEGuiRendererSelector.cpp
 
-libCEGUISampleHelper_la_LIBADD =
+libCEGUISampleHelper_la_LIBADD = -lCEGUIBase
 
 
 if CEGUI_USING_GTK2
Index: XMLParserModules/TinyXMLParser/Makefile.am
===================================================================
--- XMLParserModules/TinyXMLParser/Makefile.am	(revision 1465)
+++ XMLParserModules/TinyXMLParser/Makefile.am	(working copy)
@@ -8,6 +8,8 @@
                                    ceguitinyxml/tinyxmlerror.cpp \
                                    ceguitinyxml/tinyxmlparser.cpp
 
+libCEGUITinyXMLParser_la_LIBADD = -lCEGUIBase
+
 pkgincludedir=$(prefix)/include/CEGUI/XMLParserModules/TinyXMLParser
 pkginclude_HEADERS = CEGUITinyXMLParser.h CEGUITinyXMLParserModule.h
 
Index: XMLParserModules/libxmlParser/Makefile.am
===================================================================
--- XMLParserModules/libxmlParser/Makefile.am	(revision 1465)
+++ XMLParserModules/libxmlParser/Makefile.am	(working copy)
@@ -7,4 +7,4 @@
 pkgincludedir=$(prefix)/include/CEGUI/XMLParserModules/LibxmlParser
 pkginclude_HEADERS = CEGUILibxmlParser.h CEGUILibxmlParserModule.h
 
-libCEGUILibxmlParser_la_LIBADD = $(libxml_LIBS)
+libCEGUILibxmlParser_la_LIBADD = $(libxml_LIBS) -lCEGUIBase
Index: XMLParserModules/XercesParser/Makefile.am
===================================================================
--- XMLParserModules/XercesParser/Makefile.am	(revision 1465)
+++ XMLParserModules/XercesParser/Makefile.am	(working copy)
@@ -6,4 +6,4 @@
 pkgincludedir=$(prefix)/include/CEGUI/XMLParserModules/XercesParser
 pkginclude_HEADERS = CEGUIXercesParser.h CEGUIXercesParserModule.h
 
-libCEGUIXercesParser_la_LIBADD = $(xerces_LIBS)
+libCEGUIXercesParser_la_LIBADD = $(xerces_LIBS) -lCEGUIBase
Index: XMLParserModules/expatParser/Makefile.am
===================================================================
--- XMLParserModules/expatParser/Makefile.am	(revision 1465)
+++ XMLParserModules/expatParser/Makefile.am	(working copy)
@@ -6,4 +6,4 @@
 pkgincludedir=$(prefix)/include/CEGUI/XMLParserModules/ExpatParser
 pkginclude_HEADERS = CEGUIExpatParser.h CEGUIExpatParserModule.h
 
-libCEGUIExpatParser_la_LIBADD = $(expat_LIBS)
+libCEGUIExpatParser_la_LIBADD = $(expat_LIBS) -lCEGUIBase
Index: ImageCodecModules/CoronaImageCodec/Makefile.am
===================================================================
--- ImageCodecModules/CoronaImageCodec/Makefile.am	(revision 1465)
+++ ImageCodecModules/CoronaImageCodec/Makefile.am	(working copy)
@@ -3,6 +3,6 @@
 
 lib_LTLIBRARIES=libCEGUICoronaImageCodec.la
 libCEGUICoronaImageCodec_la_SOURCES = CEGUICoronaImageCodec.cpp CEGUICoronaImageCodecModule.cpp
-libCEGUICoronaImageCodec_la_LIBADD = $(Corona_LIBS)
+libCEGUICoronaImageCodec_la_LIBADD = $(Corona_LIBS) -lCEGUIBase
 pkginclude_HEADERS = CEGUICoronaImageCodec.h CEGUICoronaImageCodecModule.h
 
Index: ImageCodecModules/DevILImageCodec/Makefile.am
===================================================================
--- ImageCodecModules/DevILImageCodec/Makefile.am	(revision 1465)
+++ ImageCodecModules/DevILImageCodec/Makefile.am	(working copy)
@@ -4,6 +4,6 @@
 
 lib_LTLIBRARIES=libCEGUIDevILImageCodec.la
 libCEGUIDevILImageCodec_la_SOURCES = CEGUIDevILImageCodec.cpp CEGUIDevILImageCodecModule.cpp
-libCEGUIDevILImageCodec_la_LIBADD = $(DevIL_LIBS) 
+libCEGUIDevILImageCodec_la_LIBADD = $(DevIL_LIBS) -lCEGUIBase
 pkginclude_HEADERS = CEGUIDevILImageCodec.h CEGUIDevILImageCodecModule.h
 
Index: ImageCodecModules/FreeImageImageCodec/Makefile.am
===================================================================
--- ImageCodecModules/FreeImageImageCodec/Makefile.am	(revision 1465)
+++ ImageCodecModules/FreeImageImageCodec/Makefile.am	(working copy)
@@ -3,6 +3,6 @@
 
 lib_LTLIBRARIES=libCEGUIFreeImageImageCodec.la
 libCEGUIFreeImageImageCodec_la_SOURCES = CEGUIFreeImageImageCodec.cpp CEGUIFreeImageImageCodecModule.cpp
-libCEGUIFreeImageImageCodec_la_LIBADD = $(FreeImage_LIBS)
+libCEGUIFreeImageImageCodec_la_LIBADD = $(FreeImage_LIBS) -lCEGUIBase
 pkginclude_HEADERS = CEGUIFreeImageImageCodec.h CEGUIFreeImageImageCodecModule.h
 
Index: ImageCodecModules/TGAImageCodec/Makefile.am
===================================================================
--- ImageCodecModules/TGAImageCodec/Makefile.am	(revision 1465)
+++ ImageCodecModules/TGAImageCodec/Makefile.am	(working copy)
@@ -3,6 +3,7 @@
 
 lib_LTLIBRARIES=libCEGUITGAImageCodec.la
 libCEGUITGAImageCodec_la_SOURCES = CEGUITGAImageCodec.cpp CEGUITGAImageCodecModule.cpp
+libCEGUITGAImageCodec_la_LIBADD = -lCEGUIBase
 pkginclude_HEADERS = CEGUITGAImageCodec.h CEGUITGAImageCodecModule.h 
 
 
Index: ImageCodecModules/SILLYImageCodec/Makefile.am
===================================================================
--- ImageCodecModules/SILLYImageCodec/Makefile.am	(revision 1465)
+++ ImageCodecModules/SILLYImageCodec/Makefile.am	(working copy)
@@ -3,6 +3,6 @@
 
 lib_LTLIBRARIES=libCEGUISILLYImageCodec.la
 libCEGUISILLYImageCodec_la_SOURCES = CEGUISILLYImageCodec.cpp CEGUISILLYImageCodecModule.cpp 
-libCEGUISILLYImageCodec_la_LIBADD = $(SILLY_LIBS)
+libCEGUISILLYImageCodec_la_LIBADD = $(SILLY_LIBS) -lCEGUIBase
 pkginclude_HEADERS = CEGUISILLYImageCodec.h CEGUISILLYImageCodecModule.h
 
