Index: NaviDemo/Source/InputManager.cpp
===================================================================
--- NaviDemo/Source/InputManager.cpp	(revision 43)
+++ NaviDemo/Source/InputManager.cpp	(working copy)
@@ -64,13 +64,13 @@
         mInputSystem = OIS::InputManager::createInputSystem( paramList );
 
         // If possible create a buffered keyboard
-        if( mInputSystem->numKeyboards() > 0 ) {
+//       if( mInputSystem->numKeyboards() > 0 ) {
             mKeyboard = static_cast<OIS::Keyboard*>( mInputSystem->createInputObject( OIS::OISKeyboard, true ) );
             mKeyboard->setEventCallback( this );
-        }
+//        }
 
         // If possible create a buffered mouse
-        if( mInputSystem->numMice() > 0 ) {
+//        if( mInputSystem->numMice() > 0 ) {
             mMouse = static_cast<OIS::Mouse*>( mInputSystem->createInputObject( OIS::OISMouse, true ) );
             mMouse->setEventCallback( this );
 
@@ -81,10 +81,10 @@
 
             // Set mouse region
             this->setWindowExtents( width, height );
-        }
+//        }
 
         // If possible create all joysticks in buffered mode
-        if( mInputSystem->numJoySticks() > 0 ) {
+/*		if( 0 ) { //mInputSystem->numJoySticks() > 0 ) {
             mJoysticks.resize( mInputSystem->numJoySticks() );
 
             itJoystick    = mJoysticks.begin();
@@ -93,7 +93,7 @@
                 (*itJoystick) = static_cast<OIS::JoyStick*>( mInputSystem->createInputObject( OIS::OISJoyStick, true ) );
                 (*itJoystick)->setEventCallback( this );
             }
-        }
+        } */
     }
 }
 
Index: Navi/Include/NaviManager.h
===================================================================
--- Navi/Include/NaviManager.h	(revision 43)
+++ Navi/Include/NaviManager.h	(working copy)
@@ -30,6 +30,16 @@
 #include "NaviUtilities.h"
 #include "NaviSingleton.h"
 #include "WebCore.h"
+
+#ifdef min
+#undef min
+#endif
+
+#ifdef max
+#undef max
+#endif
+
+
 #include "Ogre.h"
 #include "OgrePanelOverlayElement.h"
 #include "KeyboardHook.h"
@@ -39,6 +49,7 @@
 */
 namespace NaviLibrary
 {
+    class Navi;  // need to delcare to make gccxml happy
 	/**
 	* Enumerates relative positions. Used by NaviManager::NaviPosition
 	*/
