--- Navi/Include/Navi.h	(revision 25)
+++ Navi/Include/Navi.h	(working copy)
@@ -34,7 +34,7 @@
 {
 	class _NaviExport Navi : public LLEmbeddedBrowserWindowObserver, public Ogre::WindowEventListener, public Ogre::ManualResourceLoader
 	{
-		friend NaviManager;
+		friend class NaviManager;
 
 		std::string naviName;
 		unsigned short naviWidth;
@@ -69,8 +69,8 @@
 		std::multimap<std::string, NaviDelegate> delegateMap;
 		std::multimap<std::string, NaviDelegate>::iterator delegateIter;
 		std::pair<std::multimap<std::string, NaviDelegate>::iterator, std::multimap<std::string, NaviDelegate>::iterator> dmBounds;
-		std::map<std::string, std::vector<std::string>> ensureKeysMap;
-		std::map<std::string, std::vector<std::string>>::iterator ensureKeysMapIter;
+		std::map<std::string, std::vector<std::string> > ensureKeysMap;
+		std::map<std::string, std::vector<std::string> >::iterator ensureKeysMapIter;
 		bool okayToDelete;
 		bool isVisible;
 		bool fadingOut;

--- Navi/Include/NaviMouse.h	(revision 25)
+++ Navi/Include/NaviMouse.h	(working copy)
@@ -41,7 +41,7 @@
 	class _NaviExport NaviMouse : public Singleton<NaviMouse>, public Ogre::ManualResourceLoader
 	{
 		friend class NaviManager;
-		friend NaviCursor;
+		friend class NaviCursor;
 		int mouseX, mouseY;
 		unsigned short width, height;
 		unsigned short texWidth, texHeight;

--- Navi/Include/NaviManager.h	(revision 25)
+++ Navi/Include/NaviManager.h	(working copy)
@@ -37,6 +37,7 @@
 
 namespace NaviLibrary
 {
+    class Navi; // needed for Python-Ogre gccxml processing
 	/**
 	* Enumerates relative positions. Used by NaviManager::NaviPosition
 	*/
