--- include/OGRE/OgreConfigDialog.h-old	2009-10-16 22:38:25.000000000 +1100
+++ include/OGRE/OgreConfigDialog.h	2010-01-27 13:39:41.000000000 +1100
@@ -35,8 +35,6 @@
 // Bring in the specific platform's header file: first allow forced override
 #if defined OGRE_GUI_WIN32
 # include "WIN32/OgreConfigDialogImp.h"
-#elif defined OGRE_GUI_gtk
-# include "gtk/OgreConfigDialogImp.h"
 #elif defined OGRE_GUI_GLX
 # include "GLX/OgreConfigDialogImp.h"
 #elif OGRE_PLATFORM == OGRE_PLATFORM_WIN32
Index: PythonOgreConfig_system.py
===================================================================
--- PythonOgreConfig_system.py	(revision 1074)
+++ PythonOgreConfig_system.py	(working copy)
@@ -129,7 +129,7 @@
 PATH_LIB_OPENAL=  PATH_OPENAL ## ????????????????
 
 
-PATH_INCLUDE_Ogre=          os.path.join(LOCAL_INCLUDE,'OGRE')  # os.path.join(PATH_Ogre,'OgreMain/include') 
+PATH_INCLUDE_Ogre=          os.path.join(os.path.dirname(__file__),'include/OGRE')
 PATH_INCLUDE_Ogre_Dependencies =    PATH_INCLUDE_Ogre #      os.path.join( PATH_Ogre, 'Dependencies/include')
 
 PATH_INCLUDE_OIS =          os.path.join(LOCAL_INCLUDE,'OIS') #os.path.join(PATH_OIS,'includes')    ## Note the plural include's
